app/javascript/packs/session-expire-session.ts

```diff
@@ -1,10 +1,40 @@
-const expireConfig = document.getElementById('js-expire-session');
+import { extendSession } from '@18f/identity-session';
+import type { CountdownElement } from '@18f/identity-countdown/countdown-element';
+import type { ModalElement } from '@18f/identity-modal';
 
-if (expireConfig && expireConfig.dataset.sessionTimeoutIn) {
-  const sessionTimeoutIn = parseInt(expireConfig.dataset.sessionTimeoutIn, 10) * 1000;
-  const timeoutRefreshPath = expireConfig.dataset.timeoutRefreshPath || '';
+const warningEl = document.getElementById('session-timeout-cntnr')!;
 
-  setTimeout(() => {
+const warning = Number(warningEl.dataset.warning!) * 1000;
+const sessionsURL = warningEl.dataset.sessionsUrl!;
+const sessionTimeout = Number(warningEl.dataset.sessionTimeoutIn!) * 1000;
+const modal = document.querySelector<ModalElement>('lg-modal.session-timeout-modal')!;
+const keepaliveEl = document.getElementById('session-keepalive-btn');
+const countdownEls: NodeListOf<CountdownElement> = modal.querySelectorAll('lg-countdown');
+const timeoutRefreshPath = warningEl.dataset.timeoutRefreshPath || '';
+
+let sessionExpiration = new Date(Date.now() + sessionTimeout);
+
+function showModal() {
+  modal.show();
+  countdownEls.forEach((countdownEl) => {
+    countdownEl.expiration = sessionExpiration;
+    countdownEl.start();
+  });
+}
+
+function keepalive() {
+  const isExpired = new Date(Date.now()) > sessionExpiration;
```

Comment:
This would have the same behavior if we didn't pass `Date.now()`, so should be safe to drop it:

```suggestion
  const isExpired = new Date() > sessionExpiration;
```

>When no parameters are provided, the newly-created Date object represents the current date and time as of the time of instantiation. The returned date's [timestamp](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_epoch_timestamps_and_invalid_date) is the same as the number returned by [Date.now()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#parameters

Reply:
Although I could also see an argument here to be explicit for readability's sake, so... 🤷